lib/Common/Memory/HeapInfo.h

```diff
@@ -496,7 +496,16 @@ HeapInfo::RealAlloc(Recycler * recycler, size_t sizeCat, size_t size)
 {
     Assert(HeapInfo::IsAlignedSmallObjectSize(sizeCat));
     auto& bucket = this->GetBucket<(ObjectInfoBits)(attributes & GetBlockTypeBitMask)>(sizeCat);
-    return bucket.template RealAlloc<attributes, nothrow>(recycler, sizeCat, size);
+    if (CONFIG_FLAG(ForceSoftwareWriteBarrier)
```

Comment:
Should this if-clause be in non-Release builds? Otherwise, there is a risk of a perf regression here, no? Same for the other change in this file